### PR TITLE
feat(dom): DOM - toHaveAttibute

### PR DESCRIPTION
**Pull request by fonsiher**
Added the toHaveAttribute matcher.

### CODE REVIEW

**Review by ChristianSama — APPROVED** [review]
LGTM!

File: packages/dom/test/unit/lib/ElementAssertion.test.tsx [hunk 1]
```diff
@@ -42,4 +48,63 @@ describe("[Unit] ElementAssertion.test.ts", () => {
       });
     });
   });
+  describe(".toHaveAttribute", () => {
```
Comment (JoseLion) [inline packages/dom/test/unit/lib/ElementAssertion.test.tsx:51]:
```suggestion
  });

  describe(".toHaveAttribute", () => {
```

File: packages/dom/test/unit/lib/ElementAssertion.test.tsx [hunk 2]
```diff
@@ -42,4 +48,63 @@ describe("[Unit] ElementAssertion.test.ts", () => {
...
+
+        expect(() => test.toHaveAttribute("non-existent"))
+          .toThrowError(AssertionError)
+          .toHaveMessage("Expected to have attribute \"non-existent\"");
+
+        expect(test.not.toHaveAttribute("non-existent")).toBeEqual(test);
+      });
+    });
+
+  });
```
Comment (JoseLion) [inline packages/dom/test/unit/lib/ElementAssertion.test.tsx:109]:
```suggestion
    });
  });
```

File: packages/dom/test/unit/lib/ElementAssertion.test.tsx [hunk 3]
```diff
@@ -12,6 +12,12 @@ function TestComponent(): ReactElement {
   );
 }
 
+const TestComponentElement = (): ReactElement => (
+  <button role="button" type="submit" className="btn primary" disabled>
+    click me
+  </button>
+);
```
Comment (JoseLion) [inline packages/dom/test/unit/lib/ElementAssertion.test.tsx:19]: In general, it's better to create react components as named functions.  Arrow functions will lose their name at runtime, but named functions won't. That way, the component's `displayName` is the same as the function's name.
```suggestion
function TestComponentElement(): ReactElement {
  return (
    <button role="button" type="submit" className="btn primary" disabled>
      click me
    </button>
  );
}
```

Comment (fonsiher) [reply packages/dom/test/unit/lib/ElementAssertion.test.tsx]: Thanks @JoseLion , I just addressed all your comments. Feel free to take a look at it again.

**Comment by JoseLion** [comment]
@all-contributors
please add @fonsiher for code.
please add @matycarolina for review.

**Comment by JoseLion** [comment]
@all-contributors please add @matycarolina for review.

**Pull request merged**